lgc/elfLinker/RelocHandler.cpp

```diff
@@ -69,28 +69,35 @@ static bool parseDescSetBinding(StringRef str, unsigned &descSet, unsigned &bind
 // @param [out] value : Returns value of symbol if found
 // @return : True if successful, false if not handled
 bool RelocHandler::getValue(StringRef name, uint64_t &value) {
+  llvm::errs() << "RelocHandler::getValue " << name << "\n";
+
+  auto LetterToType = [](int typeLetter) {
+    switch (typeLetter) {
+    case 's':
+      return ResourceNodeType::DescriptorSampler;
+    case 'r':
+      return ResourceNodeType::DescriptorResource;
+    case 'b':
+      return ResourceNodeType::DescriptorBuffer;
+    default:
+      return ResourceNodeType::Unknown;
+    }
+  };
 
   if (name.startswith(reloc::DescriptorOffset)) {
     // Descriptor offset in bytes in the descriptor table for its set, or in the spill table if in the root table.
     unsigned descSet = 0;
     unsigned binding = 0;
     int typeLetter = 0;
     if (parseDescSetBinding(name.drop_front(strlen(reloc::DescriptorOffset)), descSet, binding, typeLetter)) {
-      ResourceNodeType type = ResourceNodeType::Unknown;
-      switch (typeLetter) {
-      case 's':
-        type = ResourceNodeType::DescriptorSampler;
-        break;
-      case 'r':
-        type = ResourceNodeType::DescriptorResource;
-        break;
-      case 'b':
-        type = ResourceNodeType::DescriptorBuffer;
-        break;
-      }
       const ResourceNode *outerNode = nullptr;
       const ResourceNode *node = nullptr;
+      const ResourceNodeType type = LetterToType(typeLetter);
       std::tie(outerNode, node) = getPipelineState()->findResourceNode(type, descSet, binding);
+
+      if (!node)
+        node = getPipelineState()->findSingleRootResourceNode(type);
```

Comment:
I don't think this is needed.  If the resource is in top level then `(outerNode == node)`.